tests/smoke.rs

```diff
@@ -25,9 +35,18 @@ where
         if relative.parent().is_none() {
             continue;
         }
-        let decoded = percent_decode(relative.as_os_str().as_bytes())
-            .decode_utf8()
-            .unwrap();
+        #[cfg(unix)]
+        let relative = relative.as_os_str();
+        #[cfg(windows)]
+        let relative = String::from_utf16(
+            relative
+                .as_os_str()
+                .encode_wide()
+                .collect::<Vec<_>>()
+                .as_slice(),
+        )
+        .unwrap();
```

Comment:
All this bloc looks overwhelming, could not it be simpler to just do `relative.to_string_lossy().to_string().as_bytes()`?

Reply:
Really the pre-existing code here is wrong. We should never have to convert paths to strings (they should always be kept as OsStr/OsString). I'll do a separate PR to clean this up, that will simplify this part of your patch.

Reply:
:heavy_check_mark: